### ISSUE TITLE
Extra parenthesis around expression should be warned against and removed with auto-correct

### ISSUE THREAD

**Issue by donv** [bbatsov/rubocop#3426]
Rubocop should warn against an expression with an extra parenthesis around it and remove the parenthesis when auto-correct is active.  This works well for most expressions like

```ruby
t = (['Hello World!'].first)
puts t
```

which is changed to

```ruby
t = ['Hello World!'].first
puts t
```

However, Rubocop is silent if the array is indexed with brackets.

## Expected behavior

```ruby
t = (['Hello World!'][0])
puts t
```

should be changed to

```ruby
t = ['Hello World!'][0]
puts t
```

Multiple extra parenthesis should definitely be removed:

```ruby
t = ((((((['Hello World!'][0]))))))
puts t
```

should be changed to

```ruby
t = ['Hello World!']
puts t[0]
```

## Actual behavior

The scripts pass Rubocop without offences.

## Steps to reproduce the problem

Run Rubocop on the given examples.

## RuboCop version

```
$ rubocop -V
0.42.0 (using Parser 2.3.1.2, running on ruby 2.3.1 x86_64-darwin14)
```

**Comment by jonas054** [comment]
Shouldn't be too hard to fix. It's something to do with how the `[]` operator is parsed that makes us miss it, while ordinary method calls are handled without problem.
